### PR TITLE
Remove deprecated functions from OCP\User

### PR DESCRIPTION
**Pull request by rullzer**

### CODE REVIEW

File: apps/user_ldap/tests/User_LDAPTest.php [hunk 1]
```diff
@@ -489,13 +489,22 @@ public function testGetUsersSearchEmptyResult() {
 		$this->assertEquals(0, count($result));
 	}
 
+	private function getUsers($search = '', $limit = null, $offset = null) {
+		$users = \OC::$server->getUserManager()->search($search, $limit, $offset);
+		$uids = [];
+		foreach ($users as $user) {
```
Comment (ChristophWurst) [inline apps/user_ldap/tests/User_LDAPTest.php]: I know it's just a test, but `array_map` would be a nice enhancement here

File: lib/private/Encryption/Util.php [hunk 2]
```diff
@@ -271,9 +271,12 @@ public function stripPartialFileExtension($path) {
 	}
 
 	public function getUserWithAccessToMountPoint($users, $groups) {
-		$result = array();
+		$result = [];
 		if (in_array('all', $users)) {
-			$result = \OCP\User::getUsers();
+			$users = $this->userManager->search('', null, null);
+			foreach ($users as $user) {
```
Comment (ChristophWurst) [inline lib/private/Encryption/Util.php]: same as above

**Review by ChristophWurst — APPROVED** [review]
Looks good!

**Review by nickvergessen — APPROVED** [review]

File: lib/public/User.php [hunk 3]
```diff
@@ -57,31 +57,6 @@ public static function getUser() {
...
-
-	/**
-	 * Get the user display name of the user currently logged in.
-	 * @param string|null $user user id or null for current user
-	 * @return string display name
-	 * @deprecated 8.1.0 fetch \OCP\IUser (has getDisplayName()) by using method
-	 *                   get() of \OCP\IUserManager - \OC::$server->getUserManager()
-	 * @since 5.0.0
-	 */
-	public static function getDisplayName( $user = null ) {
```
Comment (MorrisJobke) [inline lib/public/User.php]: 🙈 

bildschirmfoto 2018-03-28 um 16 01 50

@rullzer Could you open tickets in those repos?

**Pull request merged**